### ISSUE TITLE
perf: simdgroup-optimized Metal matmul kernel

### ISSUE THREAD

**Issue by sauravpanda** [sauravpanda/forge-llm#157]
## Problem

The matmul_vec Metal shader uses a naive sequential dot product — each thread computes one full row with zero SIMD parallelism:

```c
for (uint j = 0; j < cols; j += 1) {
    sum += matrix[base + j] * vector[j];
}
output[id] = sum;
```

For the 576×576 Q projection: 576 threads each do 576 sequential multiply-accumulates with memory latency stalls between loads.

For the logits projection (49152×576): 193 threadgroups, each thread does 576 sequential ops = 28M sequential operations.

## Impact

**Estimated 10-20x slowdown** on matmul alone. Huge memory bandwidth underutilization.

### Option A: Simdgroup reduction
```c
float sum = 0.0f;
for (uint j = tid_in_simd; j < cols; j += 32) {
    sum += matrix[base + j] * vector[j];
}
sum = simd_sum(sum);  // Hardware warp-level reduction
if (simd_is_first()) output[row] = sum;
```

### Option B: Block tiling with shared memory (higher impact)
- Each threadgroup computes a tile of output rows
- Load vector tiles into `threadgroup` shared memory
- Use `simdgroup_matrix_multiply` (Apple's matrix hardware) for 8x8 tiles
- Process multiple rows per threadgroup for better occupancy

### Option C: Use `simdgroup_multiply_accumulate` (M3+ hardware)
Apple Silicon M3+ supports native matrix multiply instructions via `simdgroup_matrix_multiply_accumulate` — direct hardware acceleration similar to NVIDIA tensor cores.

## Expected improvement

5-10x on matmul kernel → 2-4x overall (matmul is ~60% of forward pass time)

## Files

- Codegen: `crates/forgellm-codegen-metal/src/lib.rs` (shader generation)
- Generated: `shaders/kernels.metal` lines 12-30
